src/dispatch/incident_cost/service.py

```diff
@@ -0,0 +1,197 @@
+
+
+HOURS_IN_DAY = 24
+SECONDS_IN_HOUR = 3600
+
+
+def get(*, db_session, incident_cost_id: int) -> Optional[IncidentCost]:
+    """
+    Gets an incident cost by its id.
+    """
+    return db_session.query(IncidentCost).filter(IncidentCost.id == incident_cost_id).one_or_none()
+
+
+def get_by_incident_cost_type_id(
+    *, db_session, incident_cost_type_id: int
+) -> List[Optional[IncidentCost]]:
+    """
+    Gets incident costs by incident cost type id.
+    """
+    return db_session.query(IncidentCost).filter(
+        IncidentCost.incident_cost_type_id == incident_cost_type_id
+    )
+
+
+def get_by_incident_id(*, db_session, incident_id: int) -> List[Optional[IncidentCost]]:
+    """
+    Gets incident costs by incident id.
+    """
+    return db_session.query(IncidentCost).filter(IncidentCost.incident_id == incident_id)
+
+
+def get_by_incident_id_and_incident_cost_type_id(
+    *, db_session, incident_id: int, incident_cost_type_id: int
+) -> List[Optional[IncidentCost]]:
+    """
+    Gets incident costs by incident id and incident cost type id.
+    """
+    return (
+        db_session.query(IncidentCost)
+        .filter(IncidentCost.incident_id == incident_id)
+        .filter(IncidentCost.incident_cost_type_id == incident_cost_type_id)
+    )
+
+
+def get_all(*, db_session) -> List[Optional[IncidentCost]]:
+    """
+    Gets all incident costs.
+    """
+    return db_session.query(IncidentCost)
+
+
+def create(*, db_session, incident_cost_in: IncidentCostCreate) -> IncidentCost:
+    """
+    Creates a new incident cost.
+    """
+    incident_cost = IncidentCost(**incident_cost_in.dict())
+    db_session.add(incident_cost)
+    db_session.commit()
+    return incident_cost
+
+
+def update(
+    *, db_session, incident_cost: IncidentCost, incident_cost_in: IncidentCostUpdate
+) -> IncidentCost:
+    """
+    Updates an incident cost.
+    """
+    incident_cost_data = jsonable_encoder(incident_cost)
+    update_data = incident_cost_in.dict(skip_defaults=True)
+
+    for field in incident_cost_data:
+        if field in update_data:
+            setattr(incident_cost, field, update_data[field])
+
+    db_session.add(incident_cost)
+    db_session.commit()
+    return incident_cost
+
+
+def delete(*, db_session, incident_cost_id: int):
+    """
+    Deletes an existing incident cost.
+    """
+    db_session.query(IncidentCost).filter(IncidentCost.id == incident_cost_id).delete()
+    db_session.commit()
+
+
+def get_engagement_multiplier(participant_role: str):
+    """
+    Returns an engagement multiplier for a given incident role.
+    """
+    engagement_mappings = {
+        ParticipantRoleType.incident_commander: 1,
+        ParticipantRoleType.scribe: 0.75,
+        ParticipantRoleType.liaison: 0.75,
+        ParticipantRoleType.participant: 0.5,
+        ParticipantRoleType.reporter: 0.5,
+    }
+
+    return engagement_mappings.get(participant_role)
+
+
+def calculate_incident_opportunity_cost(
```

Comment:
Do we think these can be generalized into expressions? e.g. could we define some set of incident components (participant_active_hours, incident_type, priority) and allow the user to define their own algorithm? 

my_oppertunity_cost = participant_active x incident_type x priority ?

Reply:
Yeah, I think so. I can tackle it in a separate PR.